core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -93,254 +91,126 @@ boolean compute(Consumer<OakWBuffer> computer, MemoryManager memoryManager) {
+        throw new NotImplementedException();
     }
 
     @Override
     public int limit() {
-        return value.limit();
+        throw new NotImplementedException();
     }
 
-    int limitLock() {
-        readLock.lock();
-        if (isDeleted()) {
-            readLock.unlock();
-            throw new NullPointerException();
-        }
-        int limit;
-        try {
-            limit = value.limit();
-        } finally {
-            readLock.unlock();
-        }
-        return limit;
-    }
 
     @Override
     public int remaining() {
-        return value.remaining();
+        throw new NotImplementedException();
     }
 
-    int remainingLock() {
-        readLock.lock();
-        if (isDeleted()) {
-            readLock.unlock();
-            throw new NullPointerException();
-        }
-        int remaining;
-        try {
-            remaining = value.remaining();
-        } finally {
-            readLock.unlock();
-        }
-        return remaining;
-    }
 
     @Override
     public boolean hasRemaining() {
         return value.hasRemaining();
     }
 
-    boolean hasRemainingLock() {
-        readLock.lock();
-        if (isDeleted()) {
-            readLock.unlock();
-            throw new NullPointerException();
-        }
-        boolean hasRemaining;
-        try {
-            hasRemaining = value.hasRemaining();
-        } finally {
-            readLock.unlock();
-        }
-        return hasRemaining;
-    }
-
     @Override
     public byte get(int index) {
-        return value.get(index);
-    }
-
-    byte getLock(int index) {
-        readLock.lock();
-        if (isDeleted()) {
-            readLock.unlock();
-            throw new NullPointerException();
-        }
-        byte b;
-        try {
-            b = get(index);
-        } finally {
-            readLock.unlock();
-        }
-        return b;
+        return value.get(value.position() + index);
     }
 
     @Override
     public OakWBuffer put(int index, byte b) {
         assert writeLock.isHeldByCurrentThread();
-        value.put(index, b);
+        value.put(value.position() + index, b);
         return this;
     }
 
     @Override
     // increments the position and thus should be used only in write mode
     public OakWBuffer get(byte[] dst, int offset, int length) {
         assert writeLock.isHeldByCurrentThread();
-        value.get(dst, offset, length);
+        value.get(dst, value.position() + offset, length);
         return this;
     }
 
     @Override
     public OakWBuffer put(byte[] src, int offset, int length) {
         assert writeLock.isHeldByCurrentThread();
-        value.put(src, offset, length);
+        value.put(src, value.position() + offset, length);
         return this;
     }
 
     @Override
     public OakWBuffer put(byte[] src) {
-        assert writeLock.isHeldByCurrentThread();
-        value.put(src);
-        return this;
+        throw new NotImplementedException();
```

Comment:
I suggest removing all not implemented methods from the relevant interface, instead of throwing exceptions (this relates to all `NotImplementedException()`s below).